Sunrise/AppDelegate.swift

```diff
@@ -55,6 +61,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             AppRouting.showReservationWithId(reservationId)
         }
     }
+
+    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+        deviceToken = nil
+        saveDeviceTokenForCurrentCustomer()
+    }
+
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let deviceToken = deviceToken.reduce("", {$0 + String(format: "%02X", $1)})
+        self.deviceToken = deviceToken
+        saveDeviceTokenForCurrentCustomer()
+    }
+    
+    func saveDeviceTokenForCurrentCustomer() {
+        if Commercetools.authState == .customerToken {
+            Customer.profile { result in
+                if let customerVersion = result.model?.version, result.isSuccess {
+                    var options = SetCustomTypeOptions()
+                    if let deviceToken = self.deviceToken {
+                        var type = ResourceIdentifier()
+                        type.id = "b1c3dbfb-68b9-407f-ba97-e1e0ef4c2bdb"
```

Comment:
We should use the `key` here: http://dev.commercetools.com/http-api-types.html#resourceidentifier-by-key
Otherwise it won't work across projects.

Also, if the type doesn't exist (i.e. because it was not setup on this particular project), the update would fail. We could log a debug statement here.